contrib/endpoints/src/api_manager/proto/server_config.proto

```diff
@@ -44,15 +44,13 @@ message ServerConfig {
   // The service name.
   string service_name = 8;
 
-  // Maps service configuration files to their corresponding traffic percentage.
-  // Key is the service configuration file, Value is the traffic percentage
-  // which must be greater than 0.0 and the sum must equal to 100.0.
-  map<string, double> service_configs = 12;
-
-  // TODO(jaebong): This will be deprecated once service_configs is applied
-  // The service config id to use at the start of ESP.
+  // The service config id to use at the start of ESP. It has higher precedence
+  // than init_service_configs
   string config_id = 9;
 
+  // Init Service Configs.
+  repeated ServiceConfigTrafficPercentage init_service_configs = 12;
```

Comment:
Actually, it should be:
init_service_configs first, 

config_id is not supported,  it require init_loading,   we don't support it,  it should be removed.